Edits to branchprotector-config.yaml template (#2388)
kyma-project/test-infra: templates/README.md

The template list includes:

> - `branchprotector-config.yaml` that defines configuration for branch-protection. Any changes for branch protection should be done in that file.

Reviewer: ```suggestion
- `branchprotector-config.yaml` that defines configuration for branch protection. Provide any changes related to branch protection in that file.
```
